src/extension/op_def.rs

```diff
@@ -384,6 +385,20 @@ impl OpDef {
         }
         Ok(())
     }
+
+    /// Add a lowering function to the [OpDef]
+    pub fn add_lower_func(&mut self, lower: LowerFunc) {
+        self.lower_funcs.push(lower);
```

Comment:
Why is this a vector?

Reply:
very historic at this point but the idea is that you might have different lowerings for the same op, something to revisit when we consider loweriing in anger I think